subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -818,6 +830,19 @@ public GradleExecuter withWelcomeMessageEnabled() {
         return this;
     }
 
+    @Override
+    public GradleExecuter withToolchainDetectedEnabled() {
+        disableToolchainDetection = false;
+        return this;
+    }
+
+    @Override
+    public GradleExecuter withToolchainDownloadEnabled() {
+        withToolchainDetectedEnabled();
```

Comment:
Why does enabling toolchain download also enable detection?

Reply:
Because we need the detection mechanism to detect the newly provisioned toolchain